packages/hydra-typegen/src/commands/typegen/index.ts

```diff
@@ -120,34 +149,43 @@ types don't much the metadata definiton`,
       )
     }
 
-    const config: IConfig = {
+    return {
       events,
       calls,
-      dest: path.join(process.cwd(), flags.outDir),
-      metadata: await getMetadata({
+      outDir: flags.outDir,
+      metadata: {
         source: flags.metadata,
         blockHash: flags.blockHash,
-      }),
+      },
       strict: flags.strict,
       customTypes,
-    }
-
-    this.generate(config)
+    } as IConfig
   }
 
-  generate(config: IConfig): void {
-    const { dest, customTypes } = config
+  async buildGeneratorConfig(config: IConfig): Promise<GeneratorConfig> {
+    const { outDir, customTypes } = config
 
-    debug(`Output dir: ${dest}`)
-    fs.mkdirSync(dest, { recursive: true })
+    // if (customTypes) {
```

Comment:
I needed to uncomment this code to make it work with my yaml file.

Reply:
Fixed